### PR TITLE
[Topology.Container] Update and rename methods in EdgeSetGeometryAlgorithms to compute barycentric coordinates

### PR DESCRIPTION
**Pull request by epernod**
- **Rename** compute2PointsBarycoefs -> computeEdgeBarycentricCoordinates
- **Rename** computeRest2PointsBarycoefs -> computeEdgeBarycentricCoordinates  with parameter useRestPosition to set at true
- **Add** deprecation mechanism and backward compatibility
- **Update** all methods to use sofa::geometry::Edge::getBarycentricCoordinates instead of their own wrong version

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl [hunk 1]
```diff
@@ -453,69 +437,15 @@ bool is_point_on_edge(const Vec& p, const Vec& a, const Vec& b)
...
-
-    baryCoefs.push_back(coef_a);
-    baryCoefs.push_back(coef_b);
-
-    return baryCoefs;
-
-}
-
 template<class Vec>
 sofa::type::vector< typename Vec::value_type > compute_2points_barycoefs(const Vec& p, const Vec& a, const Vec& b)
```
Comment (bakpaul) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl]: What is the use of this method ? Isn't it exactly the same as the compute2PointsBarycoefs except from the return type ? 
The fact that it does the same as the other one and its name makes me want to remove it...

Comment (epernod) [reply Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl]: ok for me, I will depreciate it as well

Comment (epernod) [reply Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl]: ah in fact it is not even a method of the class... bye bye

**Review by bakpaul — APPROVED** [review]

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl [hunk 2]
```diff
@@ -361,44 +361,28 @@ bool EdgeSetGeometryAlgorithms<DataTypes>::isPointOnEdge(const sofa::type::Vec<3
...
-    PointID ind_p2) const -> sofa::type::vector< SReal >
+    PointID ind_p1, PointID ind_p2, bool useRestPosition) const -> sofa::type::vector< SReal >
 {
-    constexpr Real ZERO = static_cast<Real>(1e-6);
-
-    sofa::type::vector< SReal > baryCoefs;
+    sofa::core::ConstVecCoordId::MyVecId _vecId = useRestPosition ? core::ConstVecCoordId::restPosition() : core::ConstVecCoordId::position();
 
-    const typename DataTypes::VecCoord& vect_c =(this->object->read(core::ConstVecCoordId::position())->getValue());
+    const typename DataTypes::VecCoord& vect_c = (this->object->read(_vecId)->getValue());
```
Comment (fredroy) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl:372]: minor suggestion but why a prefix (with _) as it is is just a variable ?

**Review by fredroy — APPROVED** [review]

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl (hunk 2 above)
Comment (epernod) [reply Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl:372]: was just to differentiate from any possible type: VecId  but I am open to any suggestion.

**Pull request merged**